examples/chart/teleport-cluster/templates/auth/statefulset.yaml

```diff
@@ -0,0 +1,278 @@
+          postStart:
+            exec:
+              command: {{ toYaml $auth.postStart.command | nindent 14 }}
+{{- end }}
+{{- if $auth.resources }}
+        resources:
+  {{- toYaml $auth.resources | nindent 10 }}
+{{- end }}
+{{- if $auth.securityContext }}
+        securityContext: {{- toYaml $auth.securityContext | nindent 10 }}
+{{- end }}
+        volumeMounts:
+{{- if $auth.enterprise }}
+        - mountPath: /var/lib/license
+          name: "license"
+          readOnly: true
+{{- end }}
+{{- if and ($auth.gcp.credentialSecretName) (eq $auth.chartMode "gcp") }}
+        - mountPath: /etc/teleport-secrets
+          name: "gcp-credentials"
+          readOnly: true
+{{- end }}
+{{- if $auth.tls.existingCASecretName }}
+        - mountPath: /etc/teleport-tls-ca
+          name: "teleport-tls-ca"
+          readOnly: true
+{{- end }}
+        - mountPath: /etc/teleport
+          name: "config"
+          readOnly: true
+        - mountPath: /var/lib/teleport
+          name: "data"
+{{- if $auth.extraVolumeMounts }}
+  {{- toYaml $auth.extraVolumeMounts | nindent 8 }}
+{{- end }}
+{{- /* Operator uses '.Values' instead of '$auth' as it will likely be moved out of the auth pods */}}
+{{- if .Values.operator.enabled }}
+      - name: "operator"
+        image: '{{ .Values.operator.image }}:{{ include "teleport-cluster.version" . }}'
+        imagePullPolicy: {{ .Values.imagePullPolicy }}
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+          initialDelaySeconds: 15
+          periodSeconds: 20
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8081
+          initialDelaySeconds: 5
+          periodSeconds: 10
+  {{- if .Values.operator.resources }}
+        resources: {{- toYaml .Values.operator.resources | nindent 10 }}
+  {{- end }}
+        volumeMounts:
+        - mountPath: /etc/teleport
+          name: "config"
+          readOnly: true
+        - mountPath: /var/lib/teleport
+          name: "data"
+{{ end }}
+      volumes:
+{{- if $auth.enterprise }}
+      - name: license
+        secret:
+          secretName: "license"
+{{- end }}
+{{- if and ($auth.gcp.credentialSecretName) (eq $auth.chartMode "gcp") }}
+      - name: gcp-credentials
+        secret:
+          secretName: {{ $auth.gcp.credentialSecretName | quote }}
+{{- end }}
+{{- if $auth.tls.existingCASecretName }}
+      - name: teleport-tls-ca
+        secret:
+          secretName: {{ $auth.tls.existingCASecretName }}
+{{- end }}
+      - name: "config"
+        configMap:
+          name: {{ .Release.Name }}-auth
+      - name: "data"
+        {{- if and ($auth.persistence.enabled) ( and (not (eq $auth.chartMode "gcp")) (not (eq $auth.chartMode "aws"))) }}
+        persistentVolumeClaim:
+          claimName: {{ if $auth.persistence.existingClaimName }}{{ $auth.persistence.existingClaimName }}{{ else }}{{ .Release.Name }}{{ end }}
+        {{- else }}
```

Comment:
If I understand correctly, the whole PR goal is to be able to mount per-pod volume to have a persistent buffer before audit log and event ingestion. In the current state, the statefulset does not currently solve this issue as it still mounts a single PVC on all pods.

AFAIU we need to remove the single PVC mount (and the single PVC) and shift to a volume template in the statefulset. We also need to explain that when switching this value you loose your existing PVC. This is not a bad thing as it will block people from shooting themselves in the foot by enabling statefulset without understanding what data is stored where.

Note: PVC volumes cannot be resized, this must be mentioned when we'll expose the PVC volume settings like disk type and size.

Reply:
Now uses a volume template